src/gt4py/next/iterator/transforms/collapse_tuple.py

```diff
@@ -25,8 +31,72 @@ def _get_tuple_size(type_: type_inference.Type) -> int:
     )
     return len(type_.dtype)
 
+def _is_let(node: ir.Node) -> bool:
+    return isinstance(node, ir.FunCall) and isinstance(node.fun, ir.Lambda)
 
-@dataclass(frozen=True)
+def _is_if_call(node: ir.Expr):
+    return isinstance(node, ir.FunCall) and node.fun == im.ref("if_")
```

Comment:
if we didn't already start this, let's start now by putting these things in a separate util file